src/main/scala/firrtl/annotations/JsonProtocol.scala

```diff
@@ -9,6 +9,20 @@ import org.json4s.native.JsonMethods._
 import org.json4s.native.Serialization
 import org.json4s.native.Serialization.{read, writePretty}
 
+trait HasSerializationHints {
+  // For serialization of complicated constuctor arguments, let the annotation
+  // writer specify additional type hints for relevant classes that might be
+  // contained within
+  def typeHints(): Seq[Class[_]]
+}
+
+/**
+  * An annotation carrying fully qualified class names for which we'd like the
+  * deserialization protocol to generate type hints for.
+  *
+  * @param typeTags the the additional class names
+  */
+case class DeserializationTypeHintsAnnotation(typeTags: Seq[String]) extends NoTargetAnnotation
```

Comment:
Is it possible to have the typeTag argument be `Seq[Class[_]]` instead of `Seq[String]`? Or does that defeat the purpose of having this annotation be serializable?

Reply:
I don't think it's serializable otherwise (but i'd have to check).

Reply:
It could be serializable by providing custom serializers!